treeherder/etl/mixins.py

```diff
@@ -48,7 +48,7 @@ class JsonLoaderMixin(object):
     """This mixin posts a json serializable object to the given url"""
 
     def load(self, url, data):
-        req = urllib2.Request(url)
+        req = urllib2.Request(url, timeout=settings.TREEHERDER_REQUESTS_TIMEOUT)
```

Comment:
Sorry missed this before - urllib2.Request() doesn't take a timeout parameter, this needs to go on the .urlopen() call a few lines further down :-)

eg:
```
[~/src]$ python
Python 2.7.9 (default, Dec 10 2014, 12:24:55) [MSC v.1500 32 bit (Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2
>>> urllib2.Request("http://www.google.com", timeout=1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'timeout'
>>>
```

Our tests must not exercise this part of the code (though I guess not surprising, since we normally have to avoid/mock out external connections like this). Longer term we should switch more things to using the Python requests package and then use the responses package to mock it out.